src/uibWinBuild.c

```diff
@@ -0,0 +1,253 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * DATE      WHO DESCRIPTION
+ * ----------------------------------------------------------------------------
+ * 08/13/20  EL  Tied the LED module, encoder, and decoder together
+ * 08/09/20  NH  Added boilerplate code
+ * 08/07/20  NH  Added serial and hardware initialization
+ * 07/02/20  NH  Initial commit
+ */
+
+/******************************************************************************
+ * Includes
+ ******************************************************************************/
+#include "compass_sim.h"
+#include "nmea.h"
+#include <stddef.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+#include "serial_sim.h"
+#include "voltage_sim.h"
+#include <pthread.h>
+#include <time.h>
+#include "status_decoder.h"
+
+/******************************************************************************
+ * Defines
+ ******************************************************************************/
+extern int LEDInit();
+extern int LEDcontrol();
+extern int decodeSensorPacket(DataStatusPacket_t* data, uint8_t* buf, uint32_t len);
+extern int sensorParse(char c);
+/******************************************************************************
+ * Typedefs
+ ******************************************************************************/
+typedef struct globalParams_
+{
+	SerialDesc_t* pGPS;
+	SerialDesc_t* pOBC;
+} globalParams_t;
+/******************************************************************************
+ * Global Data
+ ******************************************************************************/
+globalParams_t globals;
+/******************************************************************************
+ * Module Static Data
+ ******************************************************************************/
+pthread_t timer;
+/******************************************************************************
+ * Local Function Prototypes
+ ******************************************************************************/
+static int init();
+static int initSerial();
+static int initVoltage();
+static int initCompass();
+static int initTimer();
+
+static void* timerThread(void*);
+
+void app()
+{
+	uint8_t rxBuf[64];
+	int nchars;
+	while(1)
+	{
+		nchars = Serial_Read(globals.pGPS, rxBuf, 63);
+		if(nchars > 0)
+		{
+			for(int i = 0; i < nchars; i++)
+			{
+				sensorParse(rxBuf[i]);
+			}
+
+		}
+
+		nchars = Serial_Read(globals.pOBC, rxBuf, 63);
+		if(nchars > 0)
+		{
+			if(nchars == 22)
+			{
+				DataStatusPacket_t status_data;
+				decodeSensorPacket(&status_data, rxBuf, nchars);
+			}
+		}
```

Comment:
Already noted in other file, but why aren't we parsing this byte by byte?  There is no guarantee that you will get all 22 bytes in the same read.